### ISSUE TITLE
Add a SQLite function for comparing version strings

### ISSUE THREAD

**Issue by zwass** [osquery/osquery#5196]
Motivation: Users want to be able to determine whether they are running vulnerable versions of software on their systems. Many vulnerability reports include ranges of version numbers that are effected.

Implementing a `compare_version` function would allow users to do a comparison directly within their query, rather than having to do a complex splitting and comparison of values.

```
SELECT  *
FROM apps
WHERE bundle_executable = 'Google Chrome'
AND CAST(SPLIT(bundle_short_version, ".", 0) AS INT) > 67;
```
becomes
```
SELECT  *
FROM apps
WHERE bundle_executable = 'Google Chrome'
AND compare_version(bundle_short_version, '67.0.0.0') > 0
```

I can implement this if we can come to some agreement about the semantics.

**Comment by guliashvili** [comment]
That looks like a great idea. It will also be super helpful when extension vulnerabilities are discovered, in some range of the versions.
Just my 1 cent about the naming. I suppose comparing versions for equality does not need custom function. Comparing for less/more could be `bool is_version_less`

**Comment by obelisk** [comment]
If we come up with a definition this might be a good first issue task

**Comment by fmanco** [comment]
I assume we need some sort of definition of what a version is, i.e. we can't compare `20190118` with `67.0.0` or `ad984f3`.  Curiously enough @akindyakov is currently implementing code to compare versions that conform with the semantic versioning scheme. It's for a completely different context but could be directly re-used for this.

**Comment by directionless** [comment]
This was fixed in #8168
